ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/EthSigner.java

```diff
@@ -138,11 +139,17 @@ private HttpServerOptions applyConfigTlsSettingsTo(final HttpServerOptions input
       final String password = readSecretFromFile(tlsConfig.getKeyStorePasswordFile().toPath());
       result.setPfxKeyCertOptions(new PfxOptions().setPath(keyStorePathname).setPassword(password));
 
-      if (tlsConfig.getKnownClientsFile().isPresent()) {
+      if (tlsConfig.getClientAuthConstraints().isPresent()) {
+        final ClientAuthConstraints constraints = tlsConfig.getClientAuthConstraints().get();
         result.setClientAuth(ClientAuth.REQUIRED);
         try {
-          result.setTrustOptions(
-              VertxTrustOptions.whitelistClients(tlsConfig.getKnownClientsFile().get().toPath()));
+          constraints
+              .getKnownClientsFile()
+              .ifPresent(
+                  whitelistFile ->
+                      result.setTrustOptions(
+                          VertxTrustOptions.whitelistClients(
+                              whitelistFile.toPath(), constraints.allowCaAuthorisedClients())));
```

Comment:
Just a note, `allowCaAuthorisedClients -> true` will allow both CA AND self-signed clients